game/armedforces/forcegroup.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
+
+import logging
+import random
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import ClassVar, TYPE_CHECKING, Type, Any, Iterator, Optional
+
+import yaml
+from dcs import Point
+
+from game import db
+from game.data.groups import GroupRole, GroupTask
+from game.data.radar_db import UNITS_WITH_RADAR
+from game.dcs.groundunittype import GroundUnitType
+from game.dcs.shipunittype import ShipUnitType
+from game.dcs.unittype import UnitType
+from game.point_with_heading import PointWithHeading
+from game.layout.layout import TheaterLayout, AntiAirLayout, GroupLayout
+from dcs.unittype import UnitType as DcsUnitType, VehicleType, ShipType, StaticType
+
+from game.theater.theatergroup import TheaterGroup
+
+if TYPE_CHECKING:
+    from game import Game
+    from game.factions.faction import Faction
+    from game.theater import TheaterGroundObject, ControlPoint
+
+
+@dataclass
+class ForceGroup:
+    """A logical group of multiple units and layouts which have a specific tasking"""
```

Comment:
This would really benefit from some examples. From reading layouts.md, my understanding is that a `ForceGroup` can be something like `SA-10 site`, but it might also be `Generic AAA site`. Is that right?

Just to make sure I've understood the responsibility of the classes here: `ForceGroup`s are filtered by faction during campaign creation and remain stored in `ArmedForces` for future use when buying units/groups for depopulated TGOs. They don't define unit positioning, but just the set of units that are present in each grouping. Some units may be optional (like revetments), others are required (like search radars).

If I'm right you may want to copy some of that into the docstring :)